3-UsingAPIs/Week2/homework/ex3-rollAnAce.js

```diff
@@ -12,22 +12,24 @@ const rollDice = require('../../helpers/pokerDiceRoller');
 
 function rollDiceUntil(wantedValue) {
   // TODO: rewrite this function using async/await
-  const recurse = () => {
-    return rollDice().then((value) => {
-      if (value !== wantedValue) {
-        return recurse();
-      }
-      return value;
-    });
+  const recurse = async () => {
+    const value = await rollDice();
+    while (value !== wantedValue) {
+      return recurse();
+    }
+    return value;
   };
   return recurse();
```

Comment:
This is not really simplifying the function as I suggested in slack. For one thing, the body of the `while` loop is executed once only at most, because you immediately return from it. You might as well replace `while` with `if`.

Try and get rid of the `recurse` function altogether. As I posted in slack, this is a skeleton for the desired code:

```js
async function rollDiceUntil(wantedValue) {
  let value = ...
  while (value !== wantedValue) {
    ...
  }
  return value;
}
```

Reply:
Implemented